contribute/architecture/frontend-data-requests.md

```diff
@@ -0,0 +1,51 @@
+# Data requests
+
+[BackendSrv is the core data request service](https://grafana.com/docs/grafana/latest/packages_api/runtime/backendsrv) used by Grafana. This documentation highlights the high level concepts used by `BackendSrv`.
+
+## Request `cancellation`
+> **Note:** Data sources can implement their own cancellation concept. This documentation describes the `request cancellation` included in Grafana.
+
+### The challenge
+A data request can take a long time to finish. Between the request starts and finishes, the user might change context. For instance, the user navigates away or the user issues the same request again.
+
+If we allow ongoing requests to continue, it might cause unnecessary load on data sources.
+
+Grafana uses a concept referenced as `request cancellation` to cancel any ongoing request that Grafana doesn’t need.
+
+#### Before `Grafana 7.2`
+Before Grafana can cancel any data request, Grafana needs to have a way to identify each request. Grafana identifies each request with the property `requestId` [passed as options](https://github.com/grafana/grafana/blob/master/docs/sources/packages_api/runtime/backendsrvrequest.md) when you use [BackendSrv](https://grafana.com/docs/grafana/latest/packages_api/runtime/backendsrv).
+
+The cancellation logic is:
+- When an ongoing request discovers that an additional request with the same `requestId` has started, then Grafana will cancel the ongoing request.
+- When an ongoing request discovers that the “cancel all requests” `requestId`, then Grafana will cancel the ongoing request.
```

Comment:
> When an ongoing request discovers that the “cancel all requests” `requestId`

Not sure this makes sense to me. Maybe should be "When an ongoing request discovers that the special "cancel all requests" requestId was sent"? Like is "cancel all requests" some sort of marker for the backend service that everything should be canceled?

Reply:
Thank you Andrej and yes we have a special token that cancels all requests. Used when a dashboard takes a very long time to load and you click this button or navigate away :)
![image](https://user-images.githubusercontent.com/562238/95999022-c5d41c00-0e35-11eb-93e7-67435690b6d7.png)